File: src/main/java/DahRealPanda/plugins/colonyweb/colony/model/ColonySnapshot.java
package DahRealPanda.plugins.colonyweb.colony.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Full detail snapshot for a single colony.
 *
 * <p>Citizens, research and combat data live in their own cached payloads (and their own
 * endpoints) so this document stays small enough to re-fetch on every live update.</p>
 */
public class ColonySnapshot {
    public int id;
    public String name;
    public String dimension;
    public String owner;

    public List<BuilderInfo> builders = new ArrayList<>();
    public List<WorkOrderInfo> workOrders = new ArrayList<>();
    public List<BuildingInfo> buildings = new ArrayList<>();
    public Warehouse warehouse = new Warehouse();
    public ColonyStats stats = new ColonyStats();

    /** Aggregated warehouse stock for the colony. */
    public static class Warehouse {
        public boolean present;
        public List<Stack> stacks = new ArrayList<>();
    }

    /** A single aggregated stack in the warehouse. */
    public static class Stack extends ItemInfo {
        public int count;
        public int maxStackSize; // maximum items per stack for this specific item

        public Stack() {
        }

        public Stack(String itemKey, String name, int count) {
            this.itemKey = itemKey;
            this.name = name;
            this.count = count;
        }
    }
}

File: common/src/main/kotlin/DahRealPanda/plugins/colonyweb/model/ColonySnapshot.kt
package DahRealPanda.plugins.colonyweb.model

/**
 * Full detail snapshot for a single colony.
 *
 * Citizens, research and combat data live in their own cached payloads (and their own
 * endpoints) so this document stays small enough to re-fetch on every live update.
 */
data class ColonySnapshot(
    @JvmField var id: Int = 0,
    @JvmField var name: String = "",
    @JvmField var dimension: String = "",
    @JvmField var owner: String = "",
    @JvmField var builders: MutableList<BuilderInfo> = ArrayList(),
    @JvmField var workOrders: MutableList<WorkOrderInfo> = ArrayList(),
    @JvmField var buildings: MutableList<BuildingInfo> = ArrayList(),
    @JvmField var warehouse: Warehouse = Warehouse(),
    @JvmField var stats: ColonyStats = ColonyStats()
) {
    /** Aggregated warehouse stock for the colony. */
    data class Warehouse(
        @JvmField var present: Boolean = false,
        @JvmField var stacks: MutableList<Stack> = ArrayList()
    )

    /** A single aggregated stack in the warehouse. */
    data class Stack(
        @JvmField var count: Int = 0,
        @JvmField var maxStackSize: Int = 0 // maximum items per stack for this specific item
    ) : ItemInfo()
}
